x-pack/plugins/fleet/server/services/security/message_signing_service.ts

```diff
@@ -134,6 +135,22 @@ export class MessageSigningService implements MessageSigningServiceInterface {
     return publicKey;
   }
 
+  public async rotateKeyPair(): Promise<void> {
+    const isRemoved = await this.removeKeyPair();
+    if (isRemoved) {
+      await this.generateKeyPair();
+    }
+    // TODO: Apply changes to all policies
+  }
+
+  private async removeKeyPair(): Promise<boolean> {
+    const currentKeyPair = await this.getCurrentKeyPairObj();
+    if (currentKeyPair) {
+      await this.soClient.delete(MESSAGE_SIGNING_KEYS_SAVED_OBJECT_TYPE, currentKeyPair.id);
+    }
+    return true;
```

Comment:
this would always return true. should probably be false here and another return inside the `if (currentKeyPair)`.